internal/bson/document.go

```diff
@@ -136,6 +136,18 @@ func (doc *Document) Values() []any {
 
 // ReadFrom implements bsontype interface.
 func (doc *Document) ReadFrom(r *bufio.Reader) error {
+	return doc.readNested(r, 0)
+}
+
+// readNested, similarly to ReadFrom, takes raw bytes from reader
+// and unmarshal them to the Document.
+// It also takes the nesting value (if called from different bsontype),
+// and checks if the document doesn't exceed the max nesting allowed.
```

Comment:
I think this comment can be simplified. We check nesting for document and array, that's it right?